lib/Twig/Extensions/Node/Trans.php

```diff
@@ -30,15 +30,29 @@ public function compile(Twig_Compiler $compiler)
     {
         $compiler->addDebugInfo($this);
 
-        list($msg, $vars) = $this->compileString($this->getNode('body'));
+        $body = $this->getNode('body');
 
-        if (null !== $this->getNode('plural')) {
+        list($msg, $vars) = $this->compileString($body);
+
+        if (null !== $plural = $this->getNode('plural')) {
             list($msg1, $vars1) = $this->compileString($this->getNode('plural'));
 
             $vars = array_merge($vars, $vars1);
         }
 
-        $function = null === $this->getNode('plural') ? 'gettext' : 'ngettext';
+        if (null !== $context = $this->getNode('context')) {
+            $context = trim($context->getAttribute('data'));
+        }
+
+        if(!is_null($context) && !is_null($plural)){
```

Comment:
and btw, you are missing many spaces for PSR-2 compat (but this can be fixed automatically by PHP-CS-Fixer)